File: implementation/scipamato/scipamato-common-persistence-jooq/src/main/java/ch/difty/scipamato/common/persistence/FilterConditionMapper.java
package ch.difty.scipamato.common.persistence;

import java.lang.annotation.*;

import org.springframework.stereotype.Component;

/**
 * Indicates that the annotated class implements a mapper converting a generic filter into a jOOQ condition.
 */
@Target(ElementType.TYPE)
@Retention(RetentionPolicy.RUNTIME)
@Documented
@Component
public @interface FilterConditionMapper {
}

File: common/common-persistence-jooq/src/main/kotlin/ch/difty/scipamato/common/persistence/FilterConditionMapper.kt
package ch.difty.scipamato.common.persistence

import org.springframework.stereotype.Component

/**
 * Indicates that the annotated class implements a mapper converting
 * a generic filter into a jOOQ condition.
 */
@Target(AnnotationTarget.ANNOTATION_CLASS, AnnotationTarget.CLASS)
@Retention(AnnotationRetention.RUNTIME)
@MustBeDocumented
@Component
annotation class FilterConditionMapper
